core/src/peer_id.rs

```diff
@@ -164,21 +231,68 @@ impl From<PeerId> for Vec<u8> {
     }
 }
 
+#[derive(Debug)]
+pub enum PeerIdError {
+    // Variant to group errors produced by `multihash`
+    MultihashError(MhError),
+    // Variant to group errors produced by `cid`
+    CidError(CidError),
+}
+
+impl fmt::Display for PeerIdError {
```

Comment:
`libp2p-core` already imports `thiserror`. What do you think of using it instead of this manual implementation?